string-manipulation-2/src/truncate.js

```diff
@@ -0,0 +1,24 @@
+/* exported truncate */
+
+/*
+- create storage for output
+- look at each individual character one by one
+- if the length provided is less than the length of the characters, include characters
+- add to output
+- otherwise, add the full string
+- add "..." after each
+- return output
+*/
+
+function truncate(length, string) {
+  var output = '';
+  for (var i = 0; i < length; i++) {
+    if (string.length > length) {
+      output += string[i];
+    } else {
+      output = string;
+    }
```

Comment:
Sam thing here, you can do this above the loop and return so that no loop runs if it isn't needed